core/corehttp/gateway_handler.go

```diff
@@ -273,116 +275,82 @@ func (i *gatewayHandler) postHandler(w http.ResponseWriter, r *http.Request) {
 	http.Redirect(w, r, ipfsPathPrefix+k.String(), http.StatusCreated)
 }
 
-func (i *gatewayHandler) putEmptyDirHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
yup - works by virtue of core.Resolve and friends iirc. There is also a sharness test that checks against the hash.